python/servo/build_commands.py

```diff
@@ -905,6 +933,34 @@ def package_gstreamer_dlls(env, servo_exe_dir, target, uwp):
     return not missing
 
 
+def build_uwp_hololens(target, dev):
+    # determine Visual studio Build Configuration (Debug/Release) and
+    # Build Platform (x64 vs arm64)
+    vs_platforms = {
+        "x86_64": "x64",
+        "i686": "x86",
+        "aarch64": "arm64",
+    }
+    target_arch = target.split('-')[0]
+    vs_platform = vs_platforms[target_arch]
+
+    if dev:
+        Configuration = "Debug"
+    else:
+        Configuration = "Release"
+
+    print("Building Build type = {} and Platform = {}".format(Configuration, vs_platform))
+    # execute msbuild
+    # Note: /m implies to use as many CPU cores as possible while building.
+    # msbuild /m /p:project=ServoApp .\support\hololens\servoapp.sln /p:SolutionDir=.\support\hololens
+    # /p:Configuration="Debug" /p:Platform="x64" /property:AppxBundle=Always;AppxBundlePlatforms="x64"
+
+    msbuild_cmd = "msbuild /m /p:project=ServoApp .\support\hololens\servoapp.sln /p:SolutionDir=.\support\hololens "
+    msbuild_conf = "/p:Configuration=" + Configuration + " /p:Platform=" + vs_platform + \
+        " /p:AppxBundle=Always;AppxBundlePlatforms=" + vs_platform
+    check_call(msbuild_cmd + msbuild_conf)
```

Comment:
We should also pass a list instead of a string.

Reply:
Having a little bit of an issue here. If I change check_call() to:
**check_call([msbuild_cmd, msbuild_conf])**
I get the following error.  It seems line 195 has self.config["build"]["mode"] returning a ""/blank.
CalledProcessError: Command 'msbuild /m /p:project=ServoApp .\support\hololens\servoapp.sln /p:SolutionDir=.\support\hololens  /p:Configuration=Release /p:Platform=x64 /p:AppxBundle=Always;AppxBundlePlatforms=x64' returned non-zero exit status 1

  File "C:\git-mozilla\angelortiz1007-2\python\servo\build_commands.py", line 704, in build
    if build_uwp_hololens(target_triple, dev):
  File "C:\git-mozilla\angelortiz1007-2\python\servo\build_commands.py", line 961, in build_uwp_hololens
    check_call([msbuild_cmd, msbuild_conf])
  File "C:\git-mozilla\angelortiz1007-2\python\servo\command_base.py", line 195, in check_call
    raise subprocess.CalledProcessError(status, ' '.join(*args))

If I change check_call([…]) can convert the msbuild_cmd string and msbuild_conf string into a list, msbuild fails because it appears a " " is added after /p:Platform= x64 and /p:Configuration= Release.

For msbuild to work, there can't be spaces between the "=".

Any clarification or suggestions?

Reply:
Each individual argument should be its own list item: `["msbuild", "/m", "/p:project=ServoApp", ...]`

Reply:
Ok,  so I did split things up into individual list items "[…].  However, when the final command is constructed and executed, the list items are separated by a space.  This is normally a good thing.  However, in 3 of the msbuild parameters, there can't be a space between the command line option and it's value.  E.G.

/p:Configuration=Release will work, but /p:Configuration= Release will not.  I did a hybrid below in order to get msbuild to work.  Can you tell me what am I missing?

def build_uwp_hololens(target, dev):
    # determine Visual studio Build Configuration (Debug/Release) and
    # Build Platform (x64 vs arm64)
    vs_platforms = {
        "x86_64": "x64",
        "i686": "x86",
        "aarch64": "arm64",
    }
    target_arch = target.split('-')[0]
    vs_platform = vs_platforms[target_arch]

    if dev:
        Configuration = "Debug"
    else:
        Configuration = "Release"

    # execute msbuild
    # Note: /m implies to use as many CPU cores as possible while building.
    # msbuild /m /p:project=ServoApp .\support\hololens\servoapp.sln /p:SolutionDir=.\support\hololens
    # /p:Configuration="Debug" /p:Platform="x64" /property:AppxBundle=Always;AppxBundlePlatforms="x64"
    check_call(["msbuild", "/m", "/p:project=ServoApp", ".\support\hololens\servoapp.sln", **"/p:SolutionDir=.\support\hololens"**,
                **'/p:Configuration=' + Configuration**, **"/p:Platform=" + vs_platform** ])

Reply:
That looks right - any values that need to be a single string should, in fact, be a single string :)

Reply:
Done.